packages/cli/yaml/validator/src/rules/valid-oauth/__test__/valid-oauth.test.ts

```diff
@@ -71,4 +153,46 @@ describe("valid-oauth", () => {
         ];
         expect(violations).toEqual(expectedViolations);
     });
+
+    it("invalid-types", async () => {
+        const violations = await getViolationsForRule({
+            rule: ValidOauthRule,
+            absolutePathToWorkspace: join(
+                AbsoluteFilePath.of(__dirname),
+                RelativeFilePath.of("fixtures"),
+                RelativeFilePath.of("invalid"),
+                RelativeFilePath.of("types")
+            )
+        });
+        const expectedViolations: ValidationViolation[] = [
+            {
+                message: `For details on OAuth configuration, see ${OAUTH_DOCS_LINK}`,
+                nodePath: ["service", "endpoints", "getToken"],
+                relativeFilepath: RelativeFilePath.of("auth.yml"),
+                severity: "warning"
+            },
+            {
+                message:
+                    "OAuth configuration for endpoint getToken specifies 'client-id' $request.client_id, which is not a valid 'client-id' type.",
```

Comment:
Better UX for the user who hasn't specified a value is probably something along the lines of `client_id property is missing in oauth config.`